### ISSUE TITLE
getDiff() can't detect changes on Date object

### ISSUE THREAD

**Issue by rockettomatooo** [cosmicanant/recursive-diff#7]
Hi there,

I'm comparing `Date` objects with this package. Apparently, `Date` objects are objects themselves therefore `typeof new Date() === 'object'` is true. This causes the package to recurse on `Date` objects rather than comparing them. Also, this causes the package to not detect changes on `Date` objects.

Repro:
```js
import diff from 'recursive-diff'; // version 0.1.4

const obj1 = {
  date1: new Date('2019-01-01'),
  date2: new Date('2019-02-01')
};
const obj2 = {
  date1: new Date('2019-02-01'),
  date2: new Date('2019-03-01')
};

const delta = diff.getDiff(obj1, obj2); // {}
```
As far as I could follow the code, there should be a distinct handling for `Date` objects [here](https://github.com/cosmicanant/recursive-diff/blob/343dbffdc3f4087234798b083c51706061c9d7c1/index.js#L72) as it just checks for `typeof val1 === 'object'` or `val1 !== val2`.

Maybe I can try to file a pull request sometime soon but I'm not too sure.

**Comment by cosmicanant** [comment]
@Rockettomatoo thanks for reporting this issue. Will fix it.

**Comment by cosmicanant** [comment]
@Rockettomatoo this is fixed and a new version 0.2.0 has been released on npm. Please update your code accordingly.
